### DISCUSSION TITLE
Docker integration

### DISCUSSION THREAD

**Discussion by estherDow** [Camelcade/Perl5-IDEA#2432] [Q&A]
The wiki states the possibility of indexing based on a docker container, but gives little more information on the requirements on the container side. Could somebody point me to an example docker-compose.yml with a working /intellijperl mountpoint to root fs?
Many thanks in advance!

https://github.com/Camelcade/Perl5-IDEA/wiki/Docker-support

**Comment by hurricup** [comment]
Perl plugin is not aware about docker compose atm. It can work only with docker image (will create container itself).

**Comment by estherDow** [comment]
Ok understood. Thank you so much! Could you explain me where I can specify the image?

**Reply by hurricup** [reply]
when you adding interpreter, you should have a selection: local/docker

![image](https://user-images.githubusercontent.com/2811330/141452376-f2ff6006-eab5-4cba-9076-fd30a346150c.png)

Docker plugin should be enabled

**Reply by estherDow** [reply]
Thank you so much!

**Comment by estherDow** [comment]
I cannot reproduce this behavior in OSX. This might have to do with differences in how the networking connection to docker is implemented. Could you confirm that?

**Reply by hurricup** [reply]
This works only in commercial JB products with docker plugin installed

**Reply by targzeta** [reply]
It is still so? I saw that the new version of the CE has Docker supporto pinned!
![screenshot_20211116_125627](https://user-images.githubusercontent.com/2431105/141982097-c361250a-8c2c-444b-adc9-bd8641c18ca5.jpg)

**Reply by hurricup** [reply]
IRR generic docker management is available everywhere, but docker-based SDKs are only in Ultimate.

**Reply by estherDow** [reply]
I mean... fair. It's a feature I'd pay for.
